src/ui/replay_parser.rs

```diff
@@ -2366,6 +2367,192 @@ impl ToolkitTabViewer<'_> {
+                if let Some(v) = damage_received["damage_received_deep_water_torpedoes"].as_number() { stats.insert("damage_received_deep_water_torpedoes".to_string(), json!(v)); }
+                if let Some(v) = damage_received["damage_received_fire"].as_number() { stats.insert("damage_received_fire".to_string(), json!(v)); }
+                if let Some(v) = damage_received["damage_received_flooding"].as_number() { stats.insert("damage_received_flooding".to_string(), json!(v)); }
+            }
+
+            // Include raw XP if enabled
+            if ui_report.columns.contains(&ReplayColumn::RawXp) {
+                if let Some(v) = experience["raw_xp"].as_number() { stats.insert("raw_xp".to_string(), json!(v)); }
+            }
+
+            // Include distance traveled if enabled
+            if ui_report.columns.contains(&ReplayColumn::DistanceTraveled) {
+                if let Some(v) = battle_life["distance_traveled"].as_number() { stats.insert("distance_traveled".to_string(), json!(v)); }
+            }
+
+            // Include fires if enabled
+            if ui_report.columns.contains(&ReplayColumn::Fires) {
+                if let Some(v) = combat["fires_caused"].as_number() { stats.insert("fires_caused".to_string(), json!(v)); }
+            }
+
+            // Include floods if enabled
+            if ui_report.columns.contains(&ReplayColumn::Floods) {
+                if let Some(v) = combat["floods_caused"].as_number() { stats.insert("floods_caused".to_string(), json!(v)); }
+            }
+
+            // Include citadels if enabled
+            if ui_report.columns.contains(&ReplayColumn::Citadels) {
+                if let Some(v) = combat["citadels"].as_number() { stats.insert("citadels".to_string(), json!(v)); }
+            }
+
+            // Include critical hits if enabled
+            if ui_report.columns.contains(&ReplayColumn::Crits) {
+                if let Some(v) = combat["critical_hits"].as_number() { stats.insert("critical_hits".to_string(), json!(v)); }
+            }
+
+            // Build final player object
+            serde_json::json!({
+                "name": report.name_text.text(),
+                "clan": report.clan_text.as_ref().map(|clan| clan.text()),
+                "ship": report.ship_name,
+                "ship_class": report.ship_species_text,
+                "is_enemy": report.is_enemy,
+                "is_test_ship": report.is_test_ship,
+                "division": report.division_label.as_ref()
+                    .and_then(|label| label.chars().nth(1))
+                    .map(|c| c.to_string())
+                    .unwrap_or_else(|| "false".to_string()),
+                "captain_skill_points": report.skill_info.skill_points,
+                "user_id": report.vehicle.player().map(|p| p.db_id()).unwrap_or_default(),
```

Comment:
We probably want to record the realm as well. A user ID on its own is kinda useless without knowing which server it's on.